### PR TITLE
rename did to dataset_id, (and dataset.id to dataset.dataset_id), mak…

### PR DESCRIPTION
**Pull request by amueller**
…e all publish methods return self and set the id.

### CODE REVIEW

File: openml/datasets/functions.py [hunk 1]
```diff
@@ -321,9 +321,6 @@ def _get_dataset_arff(did_cache_dir, description):
     did_cache_dir : str
         Cache subdirectory for this dataset.
 
-    did : int
```
Comment (amueller) [inline openml/datasets/functions.py]: this one was wrong ;)

**Comment by amueller** [comment]
@mfeurer can you check this one out? It's not so much about the rename of did (if you don't like it I can remove it) than making the interface for the publish nicer and some minor fixes.

File: tests/datasets/test_datasets.py [hunk 2]
```diff
@@ -197,14 +196,13 @@ def test_publish_dataset(self):
         dataset = OpenMLDataset(
             name="anneal", version=1, description="test",
             format="ARFF", licence="public", default_target_attribute="class", data_file=file_path)
-        return_code, return_value = dataset.publish()
-        self.assertEqual(return_code, 200)
+        dataset.publish()
+        self.assertTrue(isinstance(dataset.dataset_id, int))
```
Comment (mfeurer) [inline tests/datasets/test_datasets.py]: Why not `assertIsInstance`?

File: tests/datasets/test_datasets.py [hunk 3]
```diff
@@ -197,14 +196,13 @@ def test_publish_dataset(self):
...
     def test_upload_dataset_with_url(self):
         dataset = OpenMLDataset(
             name="UploadTestWithURL", version=1, description="test",
             format="ARFF",
             url="http://expdb.cs.kuleuven.be/expdb/data/uci/nominal/iris.arff")
-        return_code, return_value = dataset.publish()
-        # self.assertTrue("This is a read-only account" in return_value)
-        self.assertEqual(return_code, 200)
+        dataset.publish()
+        self.assertTrue(isinstance(dataset.dataset_id, int))
```
Comment (mfeurer) [inline tests/datasets/test_datasets.py]: Why not `assertIsInstance` ?

File: tests/flows/test_flow.py [hunk 4]
```diff
@@ -20,6 +20,5 @@ def check_flow(flow):
     @unittest.skip('Not tested until test sentinels are added back.')
     def test_upload_flow(self):
         flow = openml.OpenMLFlow(model=DummyClassifier(), description="test description")
-        return_code, return_value = flow.publish()
-        # self.assertTrue("This is a read-only account" in return_value)
-        self.assertEqual(return_code, 200)
+        flow.publish()
+        self.assertTrue(isinstance(flow.flow_id, int))
```
Comment (mfeurer) [inline tests/flows/test_flow.py]: Why not `assertIsInstance`?

File: tests/runs/test_runs.py [hunk 5]
```diff
@@ -9,8 +9,7 @@ def test_run_iris(self):
         clf = LogisticRegression()
         run = openml.runs.run_task(task, clf)
         return_code, return_value = run.publish()
-        self.assertEqual(return_code, 200)
-        # self.assertTrue("This is a read-only account" in return_value)
+        self.assertTrue(isinstance(run.dataset_id, int))
```
Comment (mfeurer) [inline tests/runs/test_runs.py]: again, I think you should use a dfferent assert statement.

**Comment by mfeurer** [comment]
Okay, there are minor issues with the tests. I will patch this.

File: tests/datasets/test_datasets.py [hunk 6]
```diff
@@ -28,7 +27,7 @@ def test__list_cached_datasets(self):
         openml.config.set_cache_directory(self.static_cache_dir)
         cached_datasets = openml.datasets.functions._list_cached_datasets()
         self.assertIsInstance(cached_datasets, list)
-        self.assertEqual(len(cached_datasets), 2)
+        self.assertEqual(len(cached_datasets), 3)
```
Comment (mfeurer) [inline tests/datasets/test_datasets.py]: There is some deletion missing locally, that's why you would like to be it three locally, but two on travis-ci.

**Pull request closed without merging**